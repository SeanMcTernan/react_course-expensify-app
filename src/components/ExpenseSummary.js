import React from 'react'
import { connect } from 'react-redux'
import numeral from 'numeral'
import getVisibleExpenses from '../selectors/expenses'
import totalExpenses from '../selectors/expense-total'


export const ExpenseSummary = ({ expensesCount ,expensesTotal }) => {
    const expenseWord = expensesCount === 1 ? 'expense' : 'expenses'
    const formattedExpenseTotal = numeral(expensesTotal / 100).format('$0,0.00')
    return (
    <div>
    {
        <h1>Viewing {expensesCount} {expenseWord} totalling {formattedExpenseTotal}</h1>
      }
        </div>
)}

const mapStateToProps = (state) => {
    const visibleExpenses = getVisibleExpenses(state.expenses, state.filters) 

    return {
        expensesCount: visibleExpenses.length,
        expensesTotal: totalExpenses(visibleExpenses)
    }
} 

export default connect(mapStateToProps)(ExpenseSummary)