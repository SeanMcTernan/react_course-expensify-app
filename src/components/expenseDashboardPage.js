import React from 'react'
import ExpenseList from './ExpenseList'
import ExpenListFilters from './ExpenseListFilters' 
import ExpenseSummary  from './ExpenseSummary'

const ExpenseDashboardPage = () =>  (
    <div>
    <ExpenseSummary />
    <ExpenListFilters />
    <ExpenseList />
    </div>
)

export default ExpenseDashboardPage