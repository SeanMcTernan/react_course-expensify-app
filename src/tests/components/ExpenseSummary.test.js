import React from 'react'
import { shallow } from 'enzyme'
import { ExpenseSummary } from '../../components/ExpenseSummary'
import expenses from '../fixtures/expenses'

test('should render expense list with expenses', () => {
    const wrapper =  shallow(<ExpenseSummary expenses={expenses} />)
    expect(wrapper).toMatchSnapshot()
})

test('should render expense list with with empty message', () => {
    const wrapper =  shallow(<ExpenseSummary expenses={[]} />)
    expect(wrapper).toMatchSnapshot()
})