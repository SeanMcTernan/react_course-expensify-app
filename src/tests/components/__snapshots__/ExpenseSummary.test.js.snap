// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render expense list with expenses 1`] = `
<div>
  <p>
    Viewing 
    3
     
    expenses
     totaling 
    $1,096.95
  </p>
</div>
`;

exports[`should render expense list with with empty message 1`] = `<div />`;
